### PR TITLE
Add ability to download t-rex.exe

### PR DESCRIPTION
**Pull request by ritojo**
Downloads the miner exe

### CODE REVIEW

File: Crossed_Miner/Miner/MinerViewModel.cs [hunk 1]
```diff
@@ -72,9 +78,27 @@ private void StartMining()
...
+                            wc.DownloadProgressChanged += wc_DownloadProgressChanged;
+                            wc.DownloadFileCompleted += new AsyncCompletedEventHandler(wc_DownloadFileCompleted); ;
+                            wc.DownloadFileAsync( new System.Uri("https://github.com/trexminer/T-Rex/releases/download/0.19.9/t-rex-0.19.9-win-cuda11.1.zip"), "./t-rex.zip");
+                        }
+
+                        break;
+
+                    case MessageBoxResult.No:
+                    default:
+                        MessageBox.Show(Application.Current.MainWindow, "No mining for you then!", "Really Suck", MessageBoxButton.OK, MessageBoxImage.Error);
```
Comment (silvaresk) [inline Crossed_Miner/Miner/MinerViewModel.cs]: It should be `You Really Suck`

File: Crossed_Miner/Miner/MinerViewModel.cs [hunk 2]
```diff
@@ -72,9 +78,27 @@ private void StartMining()
...
+                MessageBoxResult result =  MessageBox.Show(Application.Current.MainWindow, "T-Rex.exe not found. Do you want to download it?", "You Suck", MessageBoxButton.YesNo, MessageBoxImage.Question);
+                switch(result)
+                {
+                    case MessageBoxResult.Yes:
+                        // Download T-Rex
+                        using (WebClient wc = new WebClient())
+                        {
+                            wc.UseDefaultCredentials = true;
+                            wc.DownloadProgressChanged += wc_DownloadProgressChanged;
+                            wc.DownloadFileCompleted += new AsyncCompletedEventHandler(wc_DownloadFileCompleted); ;
```
Comment (silvaresk) [inline Crossed_Miner/Miner/MinerViewModel.cs]: Extra `;`

File: Crossed_Miner/Miner/MinerViewModel.cs [hunk 3]
```diff
@@ -72,9 +78,27 @@ private void StartMining()
...
+                            wc.DownloadFileAsync( new System.Uri("https://github.com/trexminer/T-Rex/releases/download/0.19.9/t-rex-0.19.9-win-cuda11.1.zip"), "./t-rex.zip");
+                        }
+
+                        break;
+
+                    case MessageBoxResult.No:
+                    default:
+                        MessageBox.Show(Application.Current.MainWindow, "No mining for you then!", "Really Suck", MessageBoxButton.OK, MessageBoxImage.Error);
+                        isMining = false;
```
Comment (silvaresk) [inline Crossed_Miner/Miner/MinerViewModel.cs:99]: `IsMining` is never set to true. In `StopMining()`  it should also be set to false.

**Review by silvaresk — APPROVED** [review]
Complete

**Review by catEmperor — APPROVED** [review]

**Pull request merged**